src/lib/array.jl

```diff
@@ -469,13 +469,13 @@ using NNlib
 import NNlib: softmax, ∇softmax, logsoftmax, ∇logsoftmax, conv, ∇conv_data, depthwiseconv, maxpool, meanpool
 import NNlib: DenseConvDims, DepthwiseConvDims, PoolDims
 
-softmax(xs::TrackedArray) = track(softmax, xs)
+softmax(xs::TrackedArray; dims=1) = track(softmax, xs; dims=dims)
 
-@grad softmax(xs) = softmax(data(xs)), Δ -> (nobacksies(:softmax, ∇softmax(data(Δ), data(xs))),)
+@grad softmax(xs; dims=1) = softmax(data(xs); dims=dims), Δ -> (nobacksies(:softmax, ∇softmax(data(Δ), data(xs); dims=dims)),)
```

Comment:
There's a more efficient `∇softmax` which takes, as optional 3rd argument, the forward pass result, see here:

https://github.com/FluxML/NNlib.jl/pull/250/files#diff-858614f9f176124bafc3047ab7803e7c512f25a87f5f56661de0d3973c43a9ccR70

I think this will be in NNlib 0.6, so provided Tracker is happy to require that, it could use this (and the equivalent for logsoftmax) too.

Reply:
Shouldn't we tag NNlib with that update before making the changes in the dependent packages?

Reply:
Certainly we shouldn't tag an update to this package before that! Whether this should wait, or be done in two steps, is your call I guess.

Reply:
The latest version of NNlib broke `softmax` with Tracker (see https://github.com/FluxML/NNlib.jl/issues/251), so I would prefer if this fix could be made available as soon as possible (e.g., the bug breaks CI tests in multiple Turing packages). I just confirmed that this PR fixes the issue.